guides/doc-Provisioning_Guide/topics/proc_installing_ipv6_environment.adoc

```diff
@@ -0,0 +1,264 @@
+
+To create a IPv6 subnet via hammer do:
+
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer subnet create --name ipv6_subnet --network-type IPv6 --network 2001:db8:: --prefix 64 --tftp-id 1 --discovery-id 1 --httpboot-id 1 --template-id 1 --domains example.com
+----
+
+== Configuring Grub2 for HTTP UEFI
+
+NOTE: This step is a workaround for the following bugs in grub2 component:
++
+* https://bugzilla.redhat.com/show_bug.cgi?id=1732765
+* https://bugzilla.redhat.com/show_bug.cgi?id=1763216
+* https://bugzilla.redhat.com/show_bug.cgi?id=1616395
+
+Replace `/var/lib/tftpboot/grub2/grubx64.efi` deployed by the {foreman-installer} with a version from Fedora stable release. The file can be copied from installed Fedora EFI system or extracted from `grub2-efi` RPM package:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# wget https://download.fedoraproject.org/pub/fedora/linux/releases/31/Server/x86_64/os/Packages/g/grub2-efi-x64-2.02-100.fc31.x86_64.rpm
+# rpm2cpio grub2-efi-x64-2.02-100.fc31.x86_64.rpm | cpio -idmv
+./boot/efi/EFI/fedora/fonts
+./boot/efi/EFI/fedora/grubx64.efi
+./boot/grub2/grubenv
+./boot/loader/entries
+./etc/grub2-efi.cfg
+# cp ./boot/efi/EFI/fedora/grubx64.efi /var/lib/tftpboot/grub2/grubx64.efi
+----
+
+== Configuring external DHCPv6
+
+Both DHCPv6 and Router Advertisment services must be running on all provisioning networks. It can be installed on any host in the network, or on DHCP {SmartProxy}. {Project} currently does not manage DHCPv6 records, it is only used to bootstrap clients into installer.
+
+The DHCPv6 server must be configured to return `http://{smartproxy-example-com}:8000/EFI/grub2/grubx64.efi` URL to clients with vendor class string "HTTPClients". The URL must point to either {ProjectServer} or {SmartProxy} with TFTP and HTTPBoot features turned on. Example configuration for ISC DHCP server from {RHEL}:
+
+[options="nowrap" subs="+quotes,attributes"]
+----
+# cat /etc/dhcp/dhcpd6.conf
+default-lease-time 2592000;
+preferred-lifetime 604800;
+option dhcp-renewal-time 3600;
+option dhcp-rebinding-time 7200;
+allow leasequery;
+option dhcp6.info-refresh-time 21600;
+option dhcp6.vendor-class-identifier code 16 = string;
+option dhcp6.vendor-class code 16 = {integer 32, integer 16, string};
+option dhcp6.user-class code 15 = string;
+option dhcp6.bootfile-url code 59 = string;
+option dhcp6.client-arch-type code 61 = array of unsigned integer 16;
+
+subnet6 2001:db8::/64 {
+  range6 2001:db8::aaaa 2001:db8::ffff;
+  option dhcp6.name-servers 2001:db8::2;
+
+  if exists dhcp6.user-class and
+    substring(option dhcp6.user-class, 2, 4) = "iPXE" {
+    # iPXE script file location
+    option dhcp6.bootfile-url "http://{smartproxy-example-com}:8000/unattended/iPXE";
+  } else if option dhcp6.client-arch-type = 00:06 {
+    # UEFI x86 PXE/TFTP boot via IPv6
+    option dhcp6.bootfile-url "tftp://{smartproxy-example-com}:8000/ipxe32.efi";
```

Comment:
If we don't manage the tftp server, this won't work, right? The current example does, but keeping https://github.com/theforeman/foreman-documentation/pull/116/files#r455082819 in mind.

Reply:
Yeah these are just examples, cases we don't care about.